spec/factories/volunteers.rb

```diff
@@ -7,6 +7,15 @@
     waiver_signed true
     waiver_signed_at Time.zone.now
 
+    factory :volunteer_with_region do
+      after(:create) do |v|
+        region = create(:region)
+        v.regions << region
+        v.assigned = true
+        v.save
+      end
+    end
```

Comment:
It looks like the new factory (`volunteer_with_region`) actually does end up producing a `Volunteer` with an `Assignment`.

If we put a `binding.pry` in the test and poke around a bit, we can see there is in fact an associated `Assignment`:

```ruby
> volunteer_without_assignment.assignments.count
=> 1
```

The [`region = create(:region); v.regions << region`](https://github.com/Laszlo-JFLMTCO/food-rescue-robot/blob/feature-absences-spec/spec/factories/volunteers.rb#L12-L13) lines in the `volunteer_with_region` factory _must_ create an `Assignment` to link the `Region` to the `Volunteer`. Because the `regions` association on `Volunteer` is a [`has_many :through`](http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association), there is no way to have a `Volunteer` associated to a `Region` without an `Assignment` to join them, so those lines effectively create both a `Region` and an `Assignment` linked to that `Volunteer`.

This actually is the same outcome as the `volunteer_with_assignment` factory. There, the [`a = create(:assignment,volunteer:v); v.assignments << a`](https://github.com/Laszlo-JFLMTCO/food-rescue-robot/blob/feature-absences-spec/spec/factories/volunteers.rb#L21-L22) lines will create both a `Region` and an `Assignment` and associate them with the `Volunteer`. This is because [the `assignment` factory is set up to build an associated `Region` by default](https://github.com/Laszlo-JFLMTCO/food-rescue-robot/blob/feature-absences-spec/spec/factories/assignments.rb#L4).

---

This explains why you're not seeing the login failure error for this user. If the user truly had no assignments, they would never be able to log in.

> For this particular test case this is being used to evaluate functionaliry when a `volunteer` without any `assignment` is trying to schedule an `absence`.

Because it is not possible for a volunteer with no assignments to log into the system to get to the "Schedule an Absence" form, I think it would be best to test other cases. I think it would be best to start with the happy path: A volunteer with a shift successfully scheduling an absence for that shift.